walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -115,12 +115,13 @@
       if (mergedList.isEmpty() || !currTimeRange.overlaps(mergedList.get(mergedList.size() - 1))) {
         mergedList.add(currTimeRange);
       } else {
-        TimeRange lastTimeRange = mergedList.get(mergedList.size() - 1);
+        int mergedListSize = mergedList.size();
+        TimeRange lastTimeRange = mergedList.get(mergedListSize - 1);
         int modifiedStart = Math.min(lastTimeRange.start(), currTimeRange.start());
         int modifiedEnd = Math.max(lastTimeRange.end(), currTimeRange.end());
 
         TimeRange mergedRange = TimeRange.fromStartEnd(modifiedStart, modifiedEnd, false);
-        mergedList.remove(lastTimeRange);
+        mergedList.remove(mergedListSize - 1); // Remove with an index is constant I believe
```

Comment:
I don't think "remove with an index" is constant complexity, if that is what you mean.

Removing from the end of the list is always constant.

Reply:
OH yes got it & updated the comment thank you!!